tests/test_api.py

```diff
@@ -0,0 +1,50 @@
+import numpy as np
+
+from pygfunction.api import PYG
+
+
+def test_compute_g_functions_from_api():
+    # -- Set up
+    # borehole config
+    H = 150.  # Borehole length [m]
+    D = 4.  # Borehole buried depth [m]
+    r_b = 0.075  # Borehole radius [m]
+    x = 0.  # Borehole x-position [m]
+    y = 0.  # Borehole y-position [m]
+    tilt = np.pi / 15  # Borehole tilt [rad]
+    orientation = np.pi / 3  # Borehole orientation [rad]
+
+    # g-function config
+    alpha = 1e-6  # Ground thermal diffusivity [m2/s]
+
+    ts = H ** 2 / (9 * alpha)
+
+    time = np.array([0.1, 1., 10.]) * ts
+
+    # Set up initial borehole config values
+    complete_borehole_config = [H, D, r_b, x, y, tilt, orientation]
+    # print(f"{complete_borehole_config=}=")
+    simple_borehole_config = [H, D, r_b, x, y]
+    # print(f"{simple_borehole_config=}")
+
+    number_of_boreholes = 6
+    # only change x, y coordinates between boreholes
+    increment_indices = [3, 4]
+    # Generate list of lists of borehole configs. Increment each value by 1 to avoid duplicates
+    complete_borefield_config = [
+        [x + i if idx in increment_indices else x for idx, x in enumerate(complete_borehole_config)] for i in
+        range(number_of_boreholes)]
+    # pprint(f"{complete_borefield_config=}=")
+    simple_borefield_config = [
+        [x + i if idx in increment_indices else x for idx, x in enumerate(simple_borehole_config)] for i in
+        range(number_of_boreholes)]
+    # pprint(f"{simple_borefield_config=}=")
```

Comment:
Admittedly, I'm pretty tired this morning, but I'm not really following what all this is doing.  It feels like this is trying to set things up in a very implementation-specific way for within PyGFunction.  If so, I think it might be nice to conceal this within the API, so we can interact with it in a cleaner way.  Again -- I could totally be missing something and this is just a part of what we need to do, so just commenting.

Reply:
OK, so I was indeed a bit tired.  I thought this was having to pass indices for each borehole instance, but this is instead incrementing indices of the list of floats to create the borehole field.  I get it now, but this is a little difficult to mentally parse.  If we go ahead and add some helpers to the API file that build out boreholes from parameters, this section would also probably be cleaner.